smalltalksrc/VMMaker/CogObjectRepresentation.class.st

```diff
@@ -872,6 +872,41 @@ CogObjectRepresentation >> genPrimitiveFloatSubtract [
 	^self genDoubleArithmetic: SubRdRd preOpCheck: nil
 ]
 
+{ #category : #'primitive generators' }
+CogObjectRepresentation >> genPrimitiveFormat [
+
+	| jumpIsImm primitiveFailure |
+	"Only works for the receiver for now"
+	cogit methodNumArgs > 0 ifTrue: [ ^ UnimplementedPrimitive ].
+
+	cogit MoveR: ReceiverResultReg R: TempReg.
+
+	"Immediates don't have format"
+	cogit AndCq: objectMemory tagMask R: ReceiverResultReg.
+	jumpIsImm := cogit JumpNonZero: 0.
+	
+	cogit MoveR: ReceiverResultReg R: TempReg.
```

Comment:
Isn't this move redundant?
The live below overrides TempReg without using it...
```suggestion
```